src/kxtj3/kxtj3.hpp

```diff
@@ -0,0 +1,579 @@
+   *
+   * @return Time value in seconds
+   * @throws std::runtime_error on failure
+   */
+  float GetWakeUpMotionTime();
+
+  /**
+   * @brief Sets the timer counter of non-activity before sending another wakeup interrupt
+   *
+   * The count is limited to values from 1 to 255
+   *
+   * Every count is calculated as (1 / Wakeup_ODR_FREQUENCY) where Wakeup_ODR_FREQUENCY
+   * is the current odr_in_Hz value from odr_map_in_Hz_wakeup value
+   *
+   * Sensor must be in standby mode before performing this action
+   *
+   * See datasheet for more details
+   *
+   * @param count The timer count from 1 to 255
+   * @throws std::runtime_error on failure
+   */
+  void SetWakeUpNonActivityCounter(uint8_t count);
+
+  /**
+   * @brief Sets the timer of non-activity before sending another wakeup interrupt
+   *
+   * the desired time should be such that (0 < desired_time * wakeup_odr_frequency <= 255)
+   *
+   * Sensor must be in standby mode before performing this action
+   *
+   * See datasheet for more details
+   *
+   * @param desired_time The desired time in seconds
+   * @throws std::runtime_error on failure
+   */
+  void SetWakeUpNonActivityTime(float desired_time);
+
+  /**
+   * @brief Get the current count value of the timer of non-activity before sending another wakeup interrupt
+   *
+   * @return Time value in seconds
+   * @throws std::runtime_error on failure
+   */
+  float GetWakeUpNonActivityTime();
+
+  /**
+   * @brief Sets the threshold counter for acceleration difference before sending a wakeup interrupt
+   *
+   * The count is limited to values from 1 to 4096, that it is 16g threshold with (3.9mg/count)
+   * It is advised to not set the threshold to a very low value which may cause bad behaviour 
+   * in the wakeup interrupt
+   *
+   * Sensor must be in standby mode before performing this action
+   *
+   * See datasheet for more details
+   *
+   * @param count The timer count from 1 to 4096, that it is 16g threshold with (3.9mg/count)
+   * @throws std::runtime_error on failure
+   */
+  void SetWakeUpThresholdCounter(uint16_t count);
+
+  /**
+   * @brief Sets the threshold g value for acceleration difference before sending a wakeup interrupt
+   *
+   * The count is limited to values up to 16g, with steps of 3.9mg. It is advised to not set
+   * the threshold to a very low value which may cause bad behaviour in the wakeup interrupt
+   *
+   * Sensor must be in standby mode before performing this action
+   *
+   * See datasheet for more details
+   *
+   * @param g_threshold The acceleration threshold (in g) in g, from 3.9mg to 16g, steps of 3.9mg/count.
+   * @throws std::runtime_error on failure
+   */
+  void SetWakeUpThresholdGRange(float g_threshold);
+
+  /**
+   * @brief Get the current threshold difference value before sending wakeup interrupt
+   *
+   * @return Threshold value in g
+   * @throws std::runtime_error on failure
+   */
+  float GetWakeUpThresholdGRange();
+
+private:
```

Comment:
Since we're using the c objects, we should disable implicit copy and assignment:

```
/* Disable implicit copy and assignment operators */
KXTJ3(const KXTJ3&) = delete;
KXTJ3&operator=(const KXTJ3&) = delete;
```